### PR TITLE
[Serializer] Document Normalizers

### PR DESCRIPTION
**Pull request by lyrixx**
refs #10505

### CODE REVIEW

File: serializer/custom_normalizer.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
+.. index::
+   single: Serializer; Custom normalizers
+
+How to Create your Custom Normalizer
```
Comment (OskarStark) [inline serializer/custom_normalizer.rst]:
```suggestion
How to Create custom Normalizer
```

File: serializer/custom_normalizer.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+====================================
```
Comment (OskarStark) [inline serializer/custom_normalizer.rst]:
```suggestion
========================
```

File: serializer/custom_normalizer.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+-------------------------
+
+Imagine you want add, modify, or remove some properties during the serialization
+process. For that you'll have to create your own normalizer. But it's usually
+preferable to let Symfony normalize the object, then hook into the normalization
+to customize the normalized data. To do that, we leverage the ObjectNormalizer::
+
+    namespace AppBundle\Serializer;
+
+    use App\Entity\Topic;
```
Comment (OskarStark) [inline serializer/custom_normalizer.rst]: you use `App` here, but 2 lines above you are in `AppBundle` namespace, maybe a mistake?

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: serializer/custom_normalizer.rst (hunk 1 above)
Comment (lyrixx) [reply serializer/custom_normalizer.rst]: I copied the schema of https://symfony.com/doc/current/serializer/custom_encoders.html ; So are you sure I should be not consistant with existing documentation?

File: serializer/custom_normalizer.rst (hunk 3 above)
Comment (lyrixx) [reply serializer/custom_normalizer.rst]: yes, good catch :) thanks

File: serializer/custom_normalizer.rst (hunk 1 above)
Comment (OskarStark) [reply serializer/custom_normalizer.rst]: Fair enough, lets keep it then 👍

**Review by OskarStark — APPROVED** [review]

File: serializer/custom_normalizer.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            return $data;
+        }
+
+        public function supportsNormalization($data, $format = null)
+        {
+            return $data instanceof Topic;
+        }
+    }
+
+Registering it in your app
```
Comment (javiereguiluz) [inline serializer/custom_normalizer.rst]: A question for when we merge this into upper branches. Does Symfony autoconfiguration support Normalizers? Thanks!

Comment (lyrixx) [reply serializer/custom_normalizer.rst]: Yes it does :heart: 
I already planned to open a new PR to remove theses for new Symfony versions

**Comment by javiereguiluz** [comment]
Thanks Grégoire! This has been merged now.

**Comment by javiereguiluz** [comment]
The change it's now merged in all branches. Next steps would be:

* A PR to 3.4 to change `AppBundle` -> `App`
* Later: a PR to 4.1 to simplify things and rely on service autoconfiguration.

Thanks!

**Comment by lyrixx** [comment]
I have opened #10543 before reading your last comment.
I have documented in 3.4 the auto-configure part. It was done also for the encoder. Did I miss something ?
(let's continue this discussion if needed in #10543)

Thanks for the merge BTW

**Pull request merged**